Source/Core/InputCommon/ControllerInterface/DInput/DInputKeyboardMouse.cpp

```diff
@@ -156,38 +160,104 @@ KeyboardMouse::KeyboardMouse(const LPDIRECTINPUTDEVICE8 kb_device,
 
   // cursor, with a hax for-loop
   for (unsigned int i = 0; i < 4; ++i)
-    AddInput(new Cursor(!!(i & 2), (&m_state_in.cursor.x)[i / 2], !!(i & 1)));
+    AddInput(new Cursor(!!(i & 2), (&m_current_state.cursor.x)[i / 2], !!(i & 1)));
 
   // Raw relative mouse movement.
   for (unsigned int i = 0; i != mouse_caps.dwAxes; ++i)
   {
-    AddInput(new RelativeMouseAxis(i, false, &m_state_in.relative_mouse));
-    AddInput(new RelativeMouseAxis(i, true, &m_state_in.relative_mouse));
+    AddInput(new RelativeMouseAxis(i, false, &m_current_state.relative_mouse));
+    AddInput(new RelativeMouseAxis(i, true, &m_current_state.relative_mouse));
   }
 }
 
 void KeyboardMouse::UpdateCursorInput()
 {
-  POINT point = {};
-  GetCursorPos(&point);
-
-  // Get the cursor position relative to the upper left corner of the current window
-  // (separate or render to main)
-  ScreenToClient(s_hwnd, &point);
-
-  // Get the size of the current window (in my case Rect.top and Rect.left was zero).
-  RECT rect;
-  GetClientRect(s_hwnd, &rect);
+  POINT raw_mouse_point = {0, 0};
+  GetCursorPos(&raw_mouse_point);
+
+  // We need to handle the render window dimensions being 0. I'm using the current monitor as if it
+  // were the render window when no game is running.
+  std::pair<double, double> render_window_dimensions{
+      OctagonalMouseJail::GetInstance().GetRenderWindowWidth(),
+      OctagonalMouseJail::GetInstance().GetRenderWindowHeight()};
+  std::pair<double, double> center_of_render_window{
+      OctagonalMouseJail::GetInstance().GetCenterOfRenderWindow()};
+  if (render_window_dimensions.first == 0 || render_window_dimensions.second == 0)
```

Comment:
Noticed this while debugging, these `Get*()` functions perform a lot of redundant calculations. It's probably better to make a single `GetWindowInfo()` call that returns a struct that contains the position and dimensions -- that way your variable names are more meaningful than the `first` and `second` from the `std::pair` too.

Reply:
yeah, the `first` and `second` things are...bad. Structured bindings are a solution I found later, but I was lazy. I'll find a more meaningful way to name those, probably following your suggestion. As far as the getters go, yes, I know they're redundant. I think I was just lazy when I wrote them. I'll get around to fixing that sometime soon, probably not today though. It's one of those things that I need to be in a good mood and not tired to do.